db/mongo/client.go

```diff
@@ -47,47 +46,46 @@ func (c client) SaveEmployee(employee *models.Employee) (string, error) {
 	opts := options.Update().SetUpsert(true)
 	collection := c.conn.Database(viper.GetString(config.DbName)).Collection(collectionName)
 	if _, err := collection.UpdateOne(context.TODO(), bson.M{"id": employee.ID}, bson.M{"$set": employee}, opts); err != nil {
-		panic("oops cannot update employee")
+		return "", errors.Wrap(err, "failed to add/update employee")
 	}
-	return "", nil
+	return employee.ID, nil
 }
 
 func (c client) GetEmployeeByID(id string) (*models.Employee, error) {
 	var employee *models.Employee
 	collection := c.conn.Database(viper.GetString(config.DbName)).Collection(collectionName)
 	if err := collection.FindOne(context.TODO(), bson.M{"id": id}).Decode(&employee); err != nil {
 		if errors.Is(err, mongo.ErrNoDocuments) {
-			panic("oops error occurred ")
```

Comment:
panic should never be used anywhere except in main.

Reply:
got it sir, i removed it